src/core/cgroup.c

```diff
@@ -918,6 +918,15 @@ static void cgroup_xattr_apply(Unit *u) {
                 else
                         unit_remove_xattr_graceful(u, NULL, xn);
         }
+
+        if (u->survive)
+                unit_set_xattr_graceful(u,
+                                        /* cgroup_path= */ NULL,
+                                        "trusted.survive",
```

Comment:
this should be `user.`, so that soft reboots can also work in a container (which lacks CAP_SYS_ADMIN) or in systemd --user.

I see no reasons why code that can access cgroupfs shouldn't be able to set the flag

Reply:
Does that mean that any process will be able to configure itself as survivor, regardless of privileges? Or are there other access control mechanisms active by default?

Reply:
they will only be able to do that in case of Delegate=yes is set (i.e. when write access to cgroupfs is granted). So i wouldn't be concerned.

If you want to be extra secure with this, you could go up the tree in cgroupfs until you find a cgroup owned by root whenever you want to read the xattr, thus ignore unpriv leaves. But i personally wouldn't bother with this, write access to cgroupfs requires delegation, and that's opt-in from priviliged settings, so should not matter much.

Reply:
ah ok then it's fine, I was worried that 'user.*' would be a 'free-for-all', but if that's not the case I'll switch it over

Reply:
note that I had to add back 'trusted' as a fallback, as 'user' requires a much newer kernel, so for example CentOS8 doesn't support it and the CI was sad. But 'user' is preferred if it is available.